runelite-client/src/main/java/net/runelite/client/plugins/lmsperformancetracker/LmsPerformanceTrackerPlugin.java

```diff
@@ -0,0 +1,275 @@
+	// It's worth noting that if you aren't in a fight, all player interactions including
+	// trading & following will trigger a new fight and a new opponent. Due to this, set the lastFightTime
+	// (in FightPerformance) in the past to only be 5 seconds before the time NEW_FIGHT_DELAY would trigger
+	// and unset the opponent, in case the player follows a different player before actually starting
+	// a fight or getting attacked. In other words, remain skeptical of the validity of this event.
+	@Subscribe
+	public void onInteractingChanged(InteractingChanged event)
+	{
+		if (config.restrictToLms() && !isAtLMS())
+		{
+			return;
+		}
+
+		stopFightIfOver();
+
+		// if the client player already has a valid opponent,
+		// or the event source/target aren't players, skip any processing.
+		if ((hasOpponent() && currentFight.fightStarted()) ||
+			!(event.getSource() instanceof Player) || !(event.getTarget() instanceof Player))
+		{
+			return;
+		}
+
+		Actor opponent;
+
+		// If the event source is the player, then it is the player interacting with their potential opponent.
+		if (event.getSource() == client.getLocalPlayer())
+		{
+			opponent = event.getTarget();
+		}
+		else if (event.getTarget() == client.getLocalPlayer())
+		{
+			opponent = event.getSource();
+		}
+		else // if neither source or target was the player, skip
+		{
+			return;
+		}
+
+		// start a new fight with the new found opponent, if a new one.
+		if (!hasOpponent() ||
+			(hasOpponent() && !opponent.getName().equals(currentFight.getOpponent().getName())))
+		{
+			currentFight = new FightPerformance(client.getLocalPlayer(), (Player)opponent);
+		}
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick gameTick)
+	{
+		// hide or show panel depending if config is restricted to LMS and if player is at LMS
+		if (config.restrictToLms())
+		{
+			if (isAtLMS())
+			{
+				if (!navButtonShown && config.saveFightHistory())
+				{
+					clientToolbar.addNavigation(navButton);
+					navButtonShown = true;
+				}
+			}
+			else
+			{
+				if (navButtonShown)
+				{
+					clientToolbar.removeNavigation(navButton);
+					navButtonShown = false;
+				}
+				return; // skip all processing if not at LMS and restricted to LMS
+			}
+		}
+
+		stopFightIfOver();
+
+		if (hasOpponent())
+		{
+			{
+				// if there is an opponent, check both players' animations for attacks.
+				// If they attacked, will also add the attack to the current fight stats,
+				// taking into account the opponent's overhead prayer.
+				currentFight.checkForAttackAnimations();
+			}
+
+		}
+	}
+
+	// Returns true if the player has an opponent.
+	private boolean hasOpponent()
+	{
+		return currentFight != null;
+	}
+
+	private void stopFightIfOver()
+	{
+		if (hasOpponent() && currentFight.isFightOver())
+		{
+			stopFight();
+		}
+	}
+
+	private void stopFight()
+	{
+		// add fight to fight history if it actually started
+		if (currentFight.fightStarted())
+		{
+			panel.addFight(currentFight);
```

Comment:
Store the fights as a list in the plugin and build the panel from that. Right now they seem to be stored in the panel only.